### DISCUSSION TITLE
Defining default response

### DISCUSSION THREAD

**Discussion by konart** [goadesign/goa#3465] [Q&A]
Is  there a way to define a default response? Unless I'm missing something right now you can only have a numeric status code, but not `default`:

```yaml
      responses:
        200:
          description: Ok.
          content:
            application/json:
              schema:
                type: object
                properties:
                  success:
                    type: integer
        default:
          description: General Error
          content:
            application/json:
              schema:
                type: object
                properties:
                  error:
                    type: boolean
```

**Comment by raphael** [comment]
There isn't currently a way to define a default response. At the end of the day the code Goa generates has a limited number of possible response status codes and these are all documented. I'm wondering what the use case for `default` would be in the context of Goa?

**Reply by konart** [reply]
I'm not sure about the "in the context of Goa" part but generally speaking this is about giving a developer a chance to generate a valid openapi file that can be further used by API consumers.

The `default` response option basicall tells consumer that unless the code is 200 they should expect an error body in response.

Same goes for client generation.

**Reply by raphael** [reply]
Makes sense, Goa already generates all the possible responses including the ones used for errors. The exception would be if the generated transport code was replaced or not used or potentially a HTTP middleware that would send a different response. In all these cases though the Goa design makes it possible to describe the corresponding responses as well together with their status code. Is the use case here about using the generated OpenAPI specification without the generated server code?
